test/test_theme.rb

```diff
@@ -0,0 +1,55 @@
+require 'helper'
+
+class TestTheme < JekyllUnitTest
+  def setup
+    @theme = Theme.new('test-theme')
+    @expected_root = File.expand_path "./fixtures/test-theme", File.dirname(__FILE__)
+  end
+
+  context "initializing" do
+    should "normalize the theme name" do
+      theme = Theme.new(' Test-Theme ')
+      assert_equal "test-theme", theme.name
+    end
+
+    should "know the theme root" do
+      assert_equal @expected_root, @theme.root
+    end
+
+    should "know the theme version" do
+      assert_equal Gem::Version.new("0.1.0"), @theme.version
+    end
+
+    should "raise an error for invalid themes" do
+      assert_raises Jekyll::Errors::MissingDependencyException do
+        Theme.new("foo").version
+      end
+    end
+  end
+
+  context "path generation" do
+    [:layouts, :includes, :sass].each do |folder|
+      should "know the #{folder} path" do
+        expected = File.expand_path("_#{folder}", @expected_root)
+        assert_equal expected, @theme.public_send("#{folder}_path")
+      end
+    end
+
+    should "generate folder paths" do
+      expected = File.expand_path("./_sass", @expected_root)
+      assert_equal expected, @theme.send(:path_for, :sass)
+    end
+
+    should "not allow paths outside of the theme root" do
+      assert_equal nil, @theme.send(:path_for, "../../source")
```

Comment:
> But personally, I think Jekyll::Site and Jekyll::Theme should both extend a common class

For now, I think it's fine as is. If there's a lot of cross-over, we can patch it up later. I considered writing a filesystem handler that would basically enforce that a path was in a certain base, but a simple call to `Jekyll.sanitized_path` works just fine for that.

Anyway, the Site should be the orchestrator, the Theme is just a certain way of supplying content.